### PR TITLE
feature : alarm rules

### PR DESCRIPTION
**Pull request by wzh425**
feature : advanced rule server and front-end connection
refactor : partial reconfiguration

### CODE REVIEW

File: src/Application/Masa.Alert.Application/AlarmRules/Commands/EnabledAlarmRuleCommand.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Alert.Application.AlarmRules.Commands;
+
+public record EnabledAlarmRuleCommand(Guid AlarmRuleId) : Command
```
Comment (MayueCif) [inline src/Application/Masa.Alert.Application/AlarmRules/Commands/EnabledAlarmRuleCommand.cs:6]: DisableAlarmRuleCommand、EnabledAlarmRuleCommand merge better?

Comment (wzh425) [reply src/Application/Masa.Alert.Application/AlarmRules/Commands/EnabledAlarmRuleCommand.cs:6]: I think these are two different business behaviors, although at present they only modify the status and connect the scheduler later

File: src/Application/Masa.Alert.Application/AlarmRules/Queries/GetAlarmRuleListQuery.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Alert.Application.AlarmRules.Queries;
+
+public record GetAlarmRuleListQuery(GetAlarmRuleInputDto Input) : Query<PaginatedListDto<AlarmRuleDto>>
+{
+    public override PaginatedListDto<AlarmRuleDto> Result { get; set; } = default!;
```
Comment (MayueCif) [inline src/Application/Masa.Alert.Application/AlarmRules/Queries/GetAlarmRuleListQuery.cs:8]: new()

File: src/Domain/Masa.Alert.Domain/AlarmRules/AlarmRule.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    public int ContinuousTriggerThreshold { get; protected set; }
+
+    public SilenceCycle SilenceCycle { get; protected set; } = default!;
+
+    public List<LogMonitorItem> LogMonitorItems { get; protected set; } = new();
+
+    public ICollection<AlarmRuleItem> Items { get; protected set; } = new Collection<AlarmRuleItem>();
+
+    private AlarmRule() { }
```
Comment (MayueCif) [inline src/Domain/Masa.Alert.Domain/AlarmRules/AlarmRule.cs:33]: ?

File: src/Infrastructure/Masa.Alert.EntityFrameworkCore/Repositories/AlarmRuleEfCoreQueryableExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Alert.EntityFrameworkCore.Repositories;
+
+public static class AlarmRuleEfCoreQueryableExtensions
+{
+    public static IQueryable<AlarmRule> IncludeDetails(this IQueryable<AlarmRule> queryable, bool include = true)
+    {
+        if (!include)
```
Comment (MayueCif) [inline src/Infrastructure/Masa.Alert.EntityFrameworkCore/Repositories/AlarmRuleEfCoreQueryableExtensions.cs:10]: method name is IncludeDetails ,why include should be false

File: src/Infrastructure/Masa.Alert.Infrastructure.EntityFrameworkCore/Extensions/PaginatedList.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Masa.Alert.Infrastructure.EntityFrameworkCore.Extensions;
+
+public class PaginatedList<TEntity> where TEntity : class
+{
+    public long Total { get; set; }
+
+    public int TotalPages { get; set; }
+
+    public List<TEntity> Result { get; set; } = default!;
```
Comment (MayueCif) [inline src/Infrastructure/Masa.Alert.Infrastructure.EntityFrameworkCore/Extensions/PaginatedList.cs:9]: new()

File: src/Infrastructure/Masa.Alert.Infrastructure.EntityFrameworkCore/Extensions/PaginatedOptions.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+﻿namespace Masa.Alert.Infrastructure.EntityFrameworkCore.Extensions;
+
+public class PaginatedOptions
+{
+    public int Page { get; set; }
+
+    public int PageSize { get; set; }
+
+    public Dictionary<string, bool>? Sorting { get; set; }
```
Comment (MayueCif) [inline src/Infrastructure/Masa.Alert.Infrastructure.EntityFrameworkCore/Extensions/PaginatedOptions.cs:9]: public Dictionary<string, bool> Sorting { get; set; } = new();

File: src/Web/Masa.Alert.Web.Admin.Server/Program.cs [hunk 7]
```diff
@@ -2,15 +2,28 @@
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
 using Masa.Alert.ApiGateways.Caller.Extensions;
+using Masa.BuildingBlocks.Configuration;
```
Comment (MayueCif) [inline src/Web/Masa.Alert.Web.Admin.Server/Program.cs:5]: global using

File: src/Infrastructure/Masa.Alert.EntityFrameworkCore/Repositories/AlarmRuleEfCoreQueryableExtensions.cs (hunk 4 above)
Comment (wzh425) [reply src/Infrastructure/Masa.Alert.EntityFrameworkCore/Repositories/AlarmRuleEfCoreQueryableExtensions.cs:10]: Include is true by default

File: src/ApiGateways/Caller/Masa.Alert.ApiGateways.Caller/Services/AlarmRules/AlarmRuleService.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public async Task DeleteAsync(Guid id)
+    {
+        await DeleteAsync($"{id}");
+    }
+
+    public async Task SetIsEnabledAsync(Guid id, bool isEnabled)
+    {
+        await PutAsync($"{id}/enabled/{isEnabled}", new { });
```
Comment (MayueCif) [inline src/ApiGateways/Caller/Masa.Alert.ApiGateways.Caller/Services/AlarmRules/AlarmRuleService.cs:39]: new { }？

Comment (wzh425) [reply src/ApiGateways/Caller/Masa.Alert.ApiGateways.Caller/Services/AlarmRules/AlarmRuleService.cs:39]: Cannot pass null

**Pull request merged**